src/Workspaces/Core/Portable/PatternMatching/ContainerPatternMatcher.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.PatternMatching
+{
+    internal partial class PatternMatcher
+    {
+        private sealed partial class ContainerPatternMatcher : PatternMatcher
+        {
+            private readonly PatternSegment[] _patternSegments;
+            private readonly char[] _containerSplitCharacters;
+
+            public ContainerPatternMatcher(
+                string[] patternParts, char[] containerSplitCharacters,
+                CultureInfo culture,
+                bool allowFuzzyMatching = false)
+                : base(false, culture, allowFuzzyMatching)
+            {
+                _containerSplitCharacters = containerSplitCharacters;
+
+                _patternSegments = patternParts
+                    .Select(text => new PatternSegment(text.Trim(), allowFuzzyMatching: allowFuzzyMatching))
+                    .ToArray();
+
+                _invalidPattern = _patternSegments.Length == 0 || _patternSegments.Any(s => s.IsInvalid);
+            }
+
+            public override void Dispose()
+            {
+                base.Dispose();
```

Comment:
:memo: When you update the calls to `Dispose`, make sure to move the `base.Dispose()` calls to the end of the method instead of first.